src/XMonad/Operations.hs

```diff
@@ -438,6 +438,40 @@ setFocusX w = withWindowSet $ \ws -> do
                   currentTime
               timedEvents = [ keyPress, keyRelease, buttonPress, buttonRelease, enterNotify, leaveNotify, selectionRequest ]
 
+setNumlockMask :: X ()
+setNumlockMask = do
```

Comment:
Now that we're exporting this, maybe we could come up with a better name. Seems to me this is more of a "updateNumlockMask" or "refreshNumlockMask" or "cacheNumlockMask" than a "set" :-)

(And I wish we had a way to do handleEventHooks which ran after the core has handled the event, because then we wouln't need to export this and [customRegrabEvHook](https://github.com/TheMC47/xmonad-contrib/blob/33d1f2eccf5e606d977e5960a7b5d35e5609b1bf/XMonad/Util/Grab.hs#L87) didn't have to duplicate everything that the core handler does.)

Reply:
Good point!  I like `cacheNumlockMask` the best, I think.

> (And I wish we had a way to do handleEventHooks which ran after the core has handled the event, because then we wouln't need to export this and customRegrabEvHook didn't have to duplicate everything that the core handler does.)

Indeed :(

Reply:
There isn't anything really stopping us from adding more hooks to core… apart from this feeling that if nobody added them by now surely they can't really be that useful. 😆